<launch>
  <param name="/use_sim_time" value="false"/>
  <!-- TF: map → camera_link -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="tf_map_camera" args="0 0 0 0 0 0 map camera_link 100" /> -->
  <node pkg="tf" type="static_transform_publisher" name="tf_map_camera"
      args="-0.1 -0.3 0.1 0 1 1 1 map camera_color_optical_frame 100" />

  <!-- TF: map → world -->
  <node pkg="tf" type="static_transform_publisher" name="tf_map_world" args="0 0 0 0 0 0 map world 100" />
  
  <!-- Realsense -->
  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="align_depth" value="true" />
    <arg name="enable_pointcloud" value="true" />
  </include>

  <!-- Octomap Server -->
  <include file="$(find xarm6_octomap_avoidance)/launch/octomap_mapping.launch">
    <!-- <arg name="cloud_in" value="/camera/depth/color/points" /> -->
    <remap from="cloud_in" to="/camera/depth/color/points"/>
    <param name="frame_id" value="map" />
    <param name="resolution" value="0.015" />
    <param name="max_range" value="1.5" />
  </include>

  <!-- Octomap → MoveIt publisher -->
  <!-- <node pkg="xarm6_octomap_avoidance" type="publish_octomap_to_moveit.py" name="octomap_to_moveit" output="screen" /> -->
</launch>
